site/data/refDataApi.ts

```diff
@@ -42,6 +42,10 @@ export const fetchStops = async (input: FetchStopsInput) => {
         method: "GET",
     });
 
+    if (!res.ok) {
+        return (await res.json()) as { error: string };
```

Comment:
Another way to do this might be to use some custom errors which can then be caught by whatever's invoking this function, so this could be:

```ts
if (!res.ok) {
    if (body.error.includes("Area of polygon must be below")) {
        throw new LargePolygonError();
    }

    throw new Error(`fetchStops call failed: ${body.error}`);
}
```

where `LargePolygonError` could be defined in `site/errors/index.ts` and would look like:

```ts
export class LargePolygonError extends Error {
    constructor() {
        super();
        Object.setPrototypeOf(this, LargePolygonError.prototype);
    }
}
```

then anywhere calling it can be wrapped in a try/catch:

```ts
try {
    ...
}
catch (e) {
    if (e instanceof LargePolygonError) {
        setWarningMessage("Drawn area too big, draw a smaller area");
    } else {
        setWarningMessage("There was a problem retrieving the stops");
    }
}
```

Reply:
done